File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int soln(int n,int dp[]){
        if(n<0) return 0;
        if(n==0) return 1;
        if(dp[n]!=0) return dp[n];
        dp[n] = soln(n-1,dp)+soln(n-2,dp);
        return dp[n];
    }
    public int climbStairs(int n) {
        int dp[] = new int[n+1];
        return soln(n,dp);
    }
}

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if(n<=1) return n;
        int a = 1;
        int b = 2;
        for(int i=3;i<=n;i++){
            int t = a;
            a=b;
            b=a+t;
        }
        return b;
    }
};